extensions/amp-story/1.0/amp-story-localization-service.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {LocalizationService} from '../../../src/service/localization';
+import {Services} from '../../../src/services';
+import {registerServiceBuilderForDoc} from '../../../src/service';
+
+/**
+ * Util function to retrieve the localization service. Ensures we can retrieve
+ * the service synchronously from the amp-story codebase without running into
+ * race conditions.
+ * @param {!Element} element
+ * @return {!../../../src/service/localization.LocalizationService}
+ */
+export const getLocalizationService = (element) => {
+  let localizationService = Services.localizationForDoc(element);
+
+  if (!localizationService) {
+    localizationService = new LocalizationService(element);
+    registerServiceBuilderForDoc(element, 'localization', function () {
+      return localizationService;
+    });
```

Comment:
Nit: It's possible for multiple `LocalizationService` objects to be instantiated e.g. depending on what caller passes in for `element`. Would that result in a bug?

Reply:
Where do you see this happening? When the runtime is in the `shadow-doc` mode that supports multiple documents?

Reply:
Ignoring PWA, it should only happen if the caller does something weird like `element` is disconnected from the DOM. 

Under the normal service usage, a service wouldn't be returned. But with this setup we'd instantiate a new service.

Reply:
Got it, I don't expect the `element` to be disconnected since I'm using the `amp-story` element whenever possible. Even if we actually hit this state, I think it should be fine since we used to have a separate instances of the service (e.g. for `story-auto-ads`) before the original refactoring.